### PR TITLE
Feature/set up associations

### PR DESCRIPTION
**Pull request by jaferIdrees**
### In this branch I did the below tasks:

- Create a class Classroom with the following:

- [ ] @label instance variable, should be initialized in the constructor.
- [ ] Setter and getter for @label (remember about attr_accessor).

- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:

- [ ] Create the has-many side (a classroom has many students).
- [ ] Create the belongs-to side (a student belongs to a classroom).
- [ ] Make sure that when adding a student to a classroom it also sets the classroom for the student.
- [ ] Make sure that when setting the classroom for a student it also adds it to the classrooms' students.

- Create a class Book with the following:

- [ ] @title and @author instance variables, should be initialized in the constructor.
- [ ] Setters and getters for instance variables (remember about attr_accessor).

- Create a class Rental with the following:

- [ ] @date instance variable, should be initialized in the constructor.
- [ ] Setter and getter for @date (remember about attr_accessor).

- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:

- [ ] Create the has-many side of Book and Rental (a book has many rentals).
- [ ] Create the belongs-to side of Rental and Book (a rental belongs to a book).
- [ ] Create the has-many side of Person and Rental (a person has many rentals).
- [ ] Create the belongs-to side of Rental and Person (a rental belongs to a person).
- [ ] Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: classes/rental.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Rental class
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+  end
```
Comment (Wes-Isaac) [inline classes/rental.rb]: - [x] Great job so far, in the rental we need to create a `belongs-to` relationship between rental and book, and also rental and person.
`    book.rental << self`

File: classes/book.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (Wes-Isaac) [inline classes/book.rb]: - [x] Great job so far, in order to achieve the` has-many` association the book itself needs to be sent. Kindly fix that.

`i.e  Rental.new(....)`

File: classes/person.rb [hunk 3]
```diff
@@ -21,6 +22,10 @@ def can_use_services?
     is_of_age? || @parent_permission
   end
 
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (Wes-Isaac) [inline classes/person.rb]: - [x] Great job so far, in order to achieve the` has-many` association the person itself needs to be sent. Kindly fix that.

`i.e  Rental.new(....)`

**Review by Wes-Isaac — CHANGES_REQUESTED** [review]
Hi @jaferIdrees, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Professional README 👀.
- Great PR description 👍.
- No linter error ✅.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
-N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: classes/person.rb (hunk 3 above)
Comment (jaferIdrees) [reply classes/person.rb]: Done

File: classes/book.rb (hunk 2 above)
Comment (jaferIdrees) [reply classes/book.rb]: Done

File: classes/rental.rb (hunk 1 above)
Comment (jaferIdrees) [reply classes/rental.rb]: Done

**Review by cwaku — APPROVED** [review]
Hi @jaferIdrees 

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
 - [x] Correct GitHub Flow 👍🏾 
 - [x] Good commit history 🥇 
 - [x] Good PR title/description 💯 
 - [x] All linters passed 🥇 
 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. The previous reviewer simply missed some of them, and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**